libafl/src/events/simple.rs

```diff
@@ -220,18 +220,18 @@ where
             } => {
                 // TODO: The monitor buffer should be added on client add.
                 monitor.client_stats_insert(ClientId(0));
-                let client = monitor.client_stats_mut_for(ClientId(0));
-
-                client.update_executions(*executions, *time);
+                monitor.update_client_stats_for(ClientId(0), |client_stat| {
+                    client_stat.update_executions(*executions, *time);
+                });
 
                 monitor.display(event.name(), ClientId(0));
                 Ok(BrokerEventResult::Handled)
             }
             Event::UpdateUserStats { name, value, .. } => {
                 monitor.client_stats_insert(ClientId(0));
-                monitor
-                    .client_stats_mut_for(ClientId(0))
-                    .update_user_stats(name.clone(), value.clone());
+                monitor.update_client_stats_for(ClientId(0), |client_stat| {
+                    client_stat.update_user_stats(name.clone(), value.clone());
```

Comment:
Random question: can't we _somehow_ get around the clone here? Or make name CoW (if it isn't already)?